src/lib/components/SourceCode.svelte

```diff
@@ -0,0 +1,36 @@
+<!--
+@component
+Display source code with syntax highlighting using Highlight.js.
+See: https://github.com/metonym/svelte-highlight
+
+#### Usage:
+```tsx
+	import javascript from "svelte-highlight/languages/javascript";
+	<SourceCode language={javascript} code="console.log('Hello, world!');" />
+```
+-->
+<script lang="ts">
+	import type { Language } from 'svelte-highlight';
+	import Highlight from 'svelte-highlight';
+	import 'svelte-highlight/styles/github-dark.css';
+
+	export let language: Language;
+	export let code: string;
+</script>
```

Comment:
I optimized the import strategy
- using language-specific highlighting instead of auto (with `language` prop) - results in much less code imported
- using static imports so we get the benefits of SSR / avoid CLS
- leveraging vite's built-in CSS loading